_source/_posts/2021-09-30-oauth-sessions-with-java.adoc

```diff
@@ -0,0 +1,316 @@
+    # mimic the functionality
+    server webapp1 host.docker.internal:8081
+    server webapp2 host.docker.internal:8082
+
+----
+
+TIP: Make sure there is a trailing newline, or you may have trouble starting up HAproxy.
+
+=== Start HAproxy and Redis with Docker
+
+I'll be running the example web application directly on my laptop, but HAproxy and Redis can both run as Docker containers, create a `docker-compose.yml` file:
+
+[source,yaml]
+----
+version: '3.8'
+services:
+
+  haproxy:
+    image: docker.io/haproxy:2.4-alpine
+    volumes:
+      - ./haproxy/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg
+    extra_hosts:
+      # Docker Desktop uses `host.docker.internal` for the host,
+      # mimic this for linux installs, requires Docker 20.10+
+      - host.docker.internal:host-gateway
+    ports:
+      - 8080:8080
+
+  redis:
+    # Starts Redis without persistence
+    image: docker.io/redis:6.2.5-alpine
+    ports:
+      - 6379:6379
+----
+
+Start both HAproxy and Redis by running:
+
+[source,sh]
+----
+docker compose up
+----
+
+NOTE: You can stop the process by pressing `Ctrl`+`C` or running the `docker compose down` command from the same directory.
+
+== Build a Secure Spring Boot Application
+
+Now that the system dependencies are out of the way, let's move on to building a Spring Boot application.
+
+Create a new Spring Boot application by visiting https://start.spring.io[`start.spring.io`] and selecting the `Web` and `Okta` dependencies or by running the following command:
+
+[source,sh]
+----
+http https://start.spring.io/starter.tgz \
+  bootVersion==2.5.4 \
+  dependencies==web,okta \
+  groupId==com.example \
+  artifactId==webapp \
+  name=="Web Application" \
+  description=="Demo Web Application" \
+  packageName==com.example \
+  javaVersion==11 \
+| tar -xzvf -
+----
+
+To give a visual indication of which server handled the request, create a REST controller that displays the server port: `src/main/java/com/example/Endpoints.java`:
+
+[source,java]
+----
+package com.example;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class Endpoints {
+
+    @GetMapping("/")
+    String serverInfo(@Value("${server.port}") int port) {
+        return "Hello, your server port is: " + port;
+    }
+}
+----
+
+=== Start the Spring Boot Application
+
+In the previous section, HAproxy was bound to port `8080`, which is also the default port for Spring Boot; start the application on port `8081` using:
+
+[source,sh]
+----
+SERVER_PORT=8081 ./mvnw spring-boot:run
+----
+
+At this point, the Spring Boot application has NOT been configured to use Redis or OAuth, but you can still test out the server by grabbing the auto-generated password from the console output; it will look something like this:
```

Comment:
```suggestion
At this point, the Spring Boot application has NOT been configured to use Redis or OAuth, but you can still test out the server by grabbing the auto-generated password from the console output. It will look something like this:
```
two sentences